File: app/src/main/java/com/syncplay/DeviceListAdapter.java
package com.syncplay;

import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.BaseAdapter;
import android.widget.ListAdapter;
import android.widget.Switch;
import android.widget.TextView;

/**
 * Created by vishal on 11/5/16.
 */

class DeviceListAdapter extends BaseAdapter {
    private Activity activity;

    DeviceListAdapter(Activity activity) {
        this.activity = activity;
    }

    @Override
    public int getCount() {
        return DeviceManager.size();
    }

    @Override
    public Object getItem(int position) {
        return DeviceManager.getDevice(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    private static class ViewHolder{

        TextView textName;
        TextView textID;
        Switch attachSwitch;

    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View v = convertView;
        ViewHolder holder;

        if(convertView == null){

            v = activity.getLayoutInflater().inflate(R.layout.view_device, null);

            /****** View Holder Object to contain tabitem.xml file elements ******/

            holder = new ViewHolder();
            holder.textName = (TextView) v.findViewById(R.id.textName);
            holder.textID = (TextView)v.findViewById(R.id.textID);
            holder.attachSwitch = (Switch)v.findViewById(R.id.attachSwitch);

            /************  Set holder with LayoutInflater ************/
            v.setTag(holder);
        } else {
            holder = (ViewHolder) v.getTag();
        }

        if(DeviceManager.size() > 0)
        {
            final PlaybackDevice device = DeviceManager.getDevice(position);

            holder.textName.setText(device.getName());
            holder.textID.setText(device.getDeviceID());
            holder.attachSwitch.setChecked(device.attached());

            /******** Set Item Click Listner for LayoutInflater for each row *******/

            holder.attachSwitch.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (((Switch)v).isChecked()) {
                        device.attach();
                    } else {
                        device.deattach();
                    }
                }
            });
        }

        return v;
    }
}

File: app/src/main/java/com/syncplay/DeviceListAdapter.kt
package com.syncplay

import android.app.Activity
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.Switch
import android.widget.TextView

internal class DeviceListAdapter(private val activity: Activity) : RecyclerView.Adapter<DeviceListAdapter.ViewHolder>() {

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        internal var textName: TextView = itemView.findViewById(R.id.textName) as TextView
        internal var textID: TextView = itemView.findViewById(R.id.textID) as TextView
        internal var attachSwitch: Switch = itemView.findViewById(R.id.attachSwitch) as Switch

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder
            = ViewHolder(activity.layoutInflater.inflate(R.layout.list_item_view_device, parent, false))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val device = DeviceManager.getDevice(position)

        holder.textName.text = device.name
        holder.textID.text = device.deviceID
        holder.attachSwitch.isChecked = device.isAttached

        holder.attachSwitch.setOnClickListener { v ->
            if ((v as Switch).isChecked) {
                device.attach()
            } else {
                device.deattach()
            }
        }
    }

    override fun getItemId(position: Int): Long = position.toLong()

    override fun getItemCount(): Int = DeviceManager.size()
}
